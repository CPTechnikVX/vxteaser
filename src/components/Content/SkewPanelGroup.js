import React      from 'react';
import PropTypes  from 'prop-types';
import Constants  from '../../utils/Constants';
import classnames from 'classnames';

/**
 * Group of skewed panels
 */
export default class SkewPanelGroup extends React.PureComponent {
	static propTypes = {
		/** @ignore */
		children:  PropTypes.node,
		modifier:  PropTypes.string,
		/**
		 * width in per cent, like '45%'
		 */
		skewWidth: PropTypes.string,
	};

	render() {
		const {children, modifier, skewWidth} = this.props;
		const classList                       = [];
		const skewStyleObj                    = {
			width: skewWidth,
		};

		classList.push(Constants.ClassName.PanelGroup);

		if (modifier) {
			classList.push(modifier);
		}

		return <div className={classnames(classList)}>
			<div className={Constants.ClassName.Skew} style={skewStyleObj} />
			{children}
		</div>;
	}
}
