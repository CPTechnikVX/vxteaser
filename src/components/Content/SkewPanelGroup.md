```html
      <skewpanelgroup skewWidth="54%">
         <panel width="45%" valign="bottom">
            <row>
               <col>
                  <headline type="sub">
                     <text>Jetzt kennenlernen: Unsere</text>
                  </headline>
               </col>
            </row>
            <row>
                <col>
	               <headline type="main">
	                  <text>KATEGORIEÃœBERSICHT</text>
	               </headline>
               </col>
            </row>
         </panel>
        <panelskewbutton link="https://www.example.com/link1">Link 1</panelskewbutton>
        <panelskewbutton link="https://www.example.com/link2">Link 2</panelskewbutton>
        
         <panelskew src="https://www.example.com/link1"></panelskew>
         <panelskew src="https://www.example.com/link2"></panelskew>
      </skewpanelgroup>
```
