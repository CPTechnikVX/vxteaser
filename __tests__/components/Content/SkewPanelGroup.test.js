import React          from 'react';
import {shallow}      from 'enzyme';
import SkewPanelGroup from '../../../src/components/Content/SkewPanelGroup';

describe('render skew panel group', () => {
	test('with content', () => {
		const wrapper = shallow(
			<SkewPanelGroup modifier="first" skewWidth="55%">Content</SkewPanelGroup>
		);
		expect(wrapper).toMatchSnapshot();

		expect(wrapper.find('.vxteaser-skew').prop('style').width).toBe('55%');
	});
});
