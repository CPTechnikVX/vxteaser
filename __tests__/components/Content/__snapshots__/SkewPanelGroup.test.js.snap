// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render skew panel group with content 1`] = `
<div
  className="vxteaser-panel-group first"
>
  <div
    className="vxteaser-skew"
    style={
      Object {
        "width": "55%",
      }
    }
  />
  Content
</div>
`;
